src/core/prometheus.cc

```diff
@@ -722,7 +743,7 @@ class metric_aggregate_by_labels {
 std::string get_value_as_string(std::stringstream& s, const mi::metric_value& value) noexcept {
     std::string value_str;
     try {
-        value_str = to_str(value);
+        s << value;
```

Comment:
I think it might be best to rename this as `void write_value_as_string` and change the call sites so that the value isn't rendered twice.

Also need to fix below:
```c++
        s << "NaN";

Reply:
Yeah, definitely. Fixed.